File: chatbot.py
import telebot
from telebot import types
from database import *
import os

#token from @BotFather
bot = telebot.TeleBot("")

class User:  
    def __init__(self, user_id):
        self.user_id = user_id
        self.name = None
        self.age = None
        self.sex = None
        self.change = None

#complete
@bot.message_handler(commands=['start'])
def start(message):
    if check_user(user_id=message.from_user.id)[0]:
        me = bot.get_me()
        return
    else:
        bot.send_message(message.from_user.id, "🥀Zzzzz...")
        register(message)

#complete
def register(message):
    try:
        user = User(message.from_user.id)
        user.name = message.from_user.id
        user.age = 18
        user.sex = "Female"
        user.change = "Male and Female"
        reg_db(user_id=user.user_id,name=user.name,old=user.age,gender=user.sex,change=user.change)
    except:
        pass
    start(message)

#complete
@bot.message_handler(commands=['help'])
def help(message):
    pass

#complete
@bot.message_handler(commands=['setting'])
def setting(message):
    pass

#complete
@bot.message_handler(commands=['next'])
def next(message):
    pass

@bot.message_handler(commands=['stop'])
def stop(message):
    pass

#complete
@bot.message_handler(commands=['search'])
def search(message):
    is_open = check_open(first_id = message.from_user.id)
    if is_open[0][0]:
        #bot.register_next_step_handler(message, chat)
        bot.register_next_step_handler(message, chat)
    else:
        bot.send_message(message.from_user.id, "Looking for a partner...")
        select = select_free()
        success = False
        if not select:
            add_user(first_id = message.from_user.id)
        else:
            for sel in select:
                if check_status(first_id = message.from_user.id, second_id = sel[0] ) or message.from_user.id == sel[0]:
                    #print(message.from_user.id, "joined")
                    continue
                else:
                    #print(sel[0])
                    #print(message.from_user.id)
                    add_second_user(first_id = sel[0], second_id = message.from_user.id)
                    user_info = get_info(user_id = sel[0])
                    bot.send_message(message.from_user.id, "Partner found. You are in dialog now.")
                    user_info = get_info(user_id = message.from_user.id)
                    bot.send_message(sel[0], "Partner found. You are in dialog now.")
                    success = True
                    break
        if not success:
            time.sleep(2)
            bot.register_next_step_handler(message, search)
        else:
            bot.register_next_step_handler(message, chat)

#complete
def chat(message):
    if message.text == "/stop" or message.text == "/next":
        companion = check_companion(first_id = message.from_user.id)
        if message.text == "/next":
            bot.send_message(message.from_user.id, "You stopped the dialog. Searching for a new partner...")
        else:
            bot.send_message(message.from_user.id, "You stopped the dialog 🙄 Type /search to find a new partner")
        bot.send_message(companion, "Your partner has stopped the dialog 😔 Type /search to find a new partner")
        close_chat(first_id = message.from_user.id)
        if message.text == "/next":
            search(message)
        else:
            start(message)
        return
        
    elif not check_open(first_id=message.from_user.id)[0][0]:
        start(message)
        return
    
    companion = check_companion(first_id = message.from_user.id)
    if message.sticker:
        bot.send_sticker(companion, message.sticker.file_id)
    elif message.photo:
        file_id = None
        for item in message.photo:
            file_id = item.file_id
        bot.send_photo(companion, file_id, caption = message.caption)
    elif message.video:
        bot.send_video(companion, message.video.file_id, caption = message.caption)
    elif message.audio :
        bot.send.audio(companion, message.audio.file_id, caption = message.caption)
    elif message.voice:
        bot.send.voice(companion, message.voice.file_id)
    elif message.animation:
        bot.send_animation(companion, message.animation.file_id)
    elif message.text:
        if message.text != "/start" and message.text != "/stop" :
            if message.reply_to_message is None:
                bot.send_message(companion, message.text)
            elif message.from_user.id != message.reply_to_message.message_id:
                bot.send_message(companion, message.text, message.reply_to_message.message_id - 1)
            else:
                bot.send_message(message.chat.id, "Rejected")
        if message.text == "/search":
            bot.send_message(message.from_user.id, "You are in dialog right now 🤔")
    bot.register_next_step_handler(message, chat)


if __name__ == "__main__":
    os.system('clear')
    bot.infinity_polling()

File: chatbot.php
<?php
require_once "PHPTelebot.php";
require_once "token.php";

$bot = new PHPTelebot($apiToken);

$bot->cmd('/start', function() {
    setting();
    return 0;
});

$bot->cmd('/search', function() {
    search();
    return 0;
});

$bot->cmd('/next', function() {
    stop(1);
    search(1);
    return 0;
});

$bot->cmd('/stop', function() {
    stop();
    return 0;
});

$bot->cmd('/help', function() {
    $self['parse_mode'] = 'markdown';
    $self['disable_web_page_preview'] = true;
    Bot::sendMessage("_This bot is for anonymous chatting with strangers in Telegram. Bot can send text, stickers, voice. Do not use this bot for criminals!_", $self);
    Bot::sendMessage("_Source code: (https://github.com/Hukiooi/chatbot3)_", $self);
    
    return 0;
});

$bot->cmd('/setting', function() {
    setting();
    return 0;
});

function is_blocked(){
    $message = Bot::message();
    $id = $message['from']['id'];
    $db = new SQLite3("users.db");
    $blocked = $db->querySingle("select blocked from users where id = {$id}");
    if($blocked > 0){
        return true;
    } else {
        return false;
    }
}

function blockedMessage(){
    echo "Blocked or banned.\n";
    return 0;
}

function setting(){
    if(is_blocked()){
        blockedMessage();
    } else {
        $message = Bot::message();
        $id = $message['from']['id'];
        $db = new SQLite3("users.db");
        $checkid = $db->querySingle("select id from users where id = {$id}");
        
        if(empty($checkid) or $checkid < 1){
            $db->query("insert into users values ({$id}, 0, 0, 0)");
        }
        $self['parse_mode'] = 'markdown';
        $self['disable_web_page_preview'] = true;
        
        Bot::sendMessage("_No need set your gender 😉_", $self);
    }
    return 0;
}

function stop($params = 0){
    if(is_blocked()){
        blockedMessage();
    } else {
        $message = Bot::message();  
        $db = new SQLite3("users.db");
        
        $id = $message['from']['id'];
        $companion = $db->querySingle("select companion from users where id = {$id}");
        
        if($companion > 0){
            $data['chat_id'] = $companion;
            $db->query("update users set status = 0, companion = 0 where id = {$id}");
            $db->query("update users set status = 0, companion = 0  where id = {$companion}");
            
            if ($params > 0){
                $self['parse_mode'] = 'markdown';
                Bot::sendMessage("_You stopped the dialog. Searching for a new partner..._", $self);
            } else {
                Bot::sendMessage("You stopped the dialog 🙄 \nType /search to find a new partner");
            }
            Bot::sendMessage("Your partner has stopped the dialog 😞 \nType /search to find a new partner", $data);
        }
        if($params < 1 and $companion < 1){
            $db->query("update users set status = 0, companion = 0 where id = {$id}");
            Bot::sendMessage("You have no partner 🤔 \nType /search to find a new partner");
        }
    }
    return 0;
}

function search($params = 0){
    if(is_blocked()){
        blockedMessage();
    } else {
        $message = Bot::message();
        $db = new SQLite3("users.db");
        
        $id = $message['from']['id'];
        $check_companion = $db->querySingle("select companion from users where status = 2 and id = {$id} limit 1");
        if($check_companion < 1){
            $db->query("update users set status = 1 where id = {$id}");
            $results = $db->querySingle("select id from users where status = 1 and id != {$id} limit 1");
            $companion = $results;
            
            if($companion > 0){        
                $data['chat_id'] = $companion;
                $db->query("update users set status = 2, companion = {$companion} where id = {$id}");
                $db->query("update users set status = 2, companion = {$id} where id = {$companion}");
                
                Bot::sendMessage("Partner found 🐵 \n/next — find a new partner \n/stop — stop this dialog");
                Bot::sendMessage("Partner found 🐵 \n/next — find a new partner \n/stop — stop this dialog", $data);
            }
            if($params < 1 and $companion < 1){
                $self['parse_mode'] = 'markdown';
                Bot::sendMessage("Looking for a partner...", $self);
            }
        } else {
            Bot::sendMessage("You are in the dialog right now 🤔 \n/next — find a new partner \n/stop — stop this dialog");
        }
    }
    return 0;
}

$bot->cmd('*', function($text){
    if(is_blocked()){
        blockedMessage();
    } else {
        $message = Bot::message();
        $id = $message['from']['id'];  
        $db = new SQLite3("users.db");
        $companion = $db->querySingle("select companion from users where id = {$id} and status = 2 limit 1");
        $data['chat_id'] = $companion;
        
        if ($companion > 0){       
            if (is_string($text)){
                Bot::sendMessage($text, $data);
            }
        }
    }
    return 0;
});

$bot->on('*', function($message){
    if(is_blocked()){
        blockedMessage();
    } else {
        $message = Bot::message();
        $id = $message['from']['id'];  
        $db = new SQLite3("users.db");
        $companion = $db->querySingle("select companion from users where id = {$id} and status = 2 limit 1");
        $data['chat_id'] = $companion;
        
        if ($companion > 0){
            if(@$message['sticker']){
                @$sticker = $message['sticker']['file_id'];
                Bot::sendSticker($sticker, $data);
            }
            if(@$message['voice']){
                $voice = $message['voice']['file_id'];
                Bot::sendVoice($voice, $data);
            }
            
        }
    }
    return 0;
});

$bot->run();
